trellis-triplestore/src/main/java/org/trellisldp/triplestore/TriplestoreResourceService.java

```diff
@@ -577,7 +587,9 @@ private void init() {
 
     @Override
     public Optional<Resource> get(final IRI identifier, final Instant time) {
-        return mementoService.map(svc -> svc.get(identifier, time)).orElseGet(() -> get(identifier));
+        // TODO -- JDK9 replace with Optional::or
+        final Optional<Resource> res = mementoService.flatMap(svc -> svc.get(identifier, time));
+        return res.isPresent() ? res : get(identifier);
```

Comment:
I am wondering about this function. Rather than checking `res.isPresent()`, I think a better approach would be:

```java
public Optional<Resource> get(final IRI identifier, final Instant time) {
    return mementoService.isPresent() ? mementoService.flatMap(svc -> svc.get(identifier, time)) : get(identifier);
}
```

This way, if the memento service isn't present, it defaults to a simple `::get`, but if the memento service _is_ present, and a client requests a datetime the precedes the creation of the resource, then an empty `Optional` is returned, which is what I'd expect (i.e. a 404 for requests that predate the resource). Right now the code returns the current state of the resource if no particular memento is found.

Reply:
WTF? I did this change (I agree with your point about the semantics of time-sensitive`::get`, but now the app tests are failing... can you see that? I was going to do this and merge, but hm...

Reply:
You'll want to revert this last change (i.e. leaving the time-sensitive `::get` as it was) -- there are a number of interrelated issues, beginning with some code in the HTTP layer, and it would be better to simply merge all of this and deal with those changes at once.

Reply:
Can do. Let's land this darn thing (I have let it fly for way too long) and charge forward.